### PR TITLE
1363 auto correct form fields

### LINKED ISSUE #1363: Code auto-correct missing periods and capitalizations
### Overview
The LG has some instances where the user should write sentences. The user may or may not add a period at the end of the sentence. In all the spaces users enter words, they may or may not capitalize the first word of a sentence (or possibly proper nouns). 

This issue is to build automatic detection for these instances, and to automatically correct them. 

### Action Items
- [x] Code a solution that identifies and fixes sentences missing capitalization or ending periods.

### Resources/Notes
- In the LG flow, we'd want this to execute after the user enters the answer, but before the text preview.

### PR DESCRIPTION
**Pull request by davidwiese**
Fixes #1363 

### Changes made
Updated all form inputs to properly capitalize or add punctuation as necessary within the context of the generated form. Tested edge cases as I encountered or thought of them.

### Reason for changes
This makes the LG more user friendly, and will require the user to make fewer edits on the preview pages. It will also help prevent their letter from looking unpolished.

### CODE REVIEW

**Review by sydneywalcoff — CHANGES_REQUESTED** [review]
@davidwiese firstly, thank you for fixing that variable typo lol.

secondly, since a lot of the code to check for capitalization/punctuation is the same, can we refactor it into a reusable helper function?

**Comment by davidwiese** [comment]
@sydneywalcoff 

Made the changes where able

File: products/statement-generator/src/helpers/statementGenerators.ts [hunk 1]
```diff
@@ -40,7 +40,10 @@ export function generateInvolvementJob(formState: IStepState): string {
     return '';
   }
 
-  return `I have been working at ${companyName} as a ${jobTitle}. At ${companyName}, ${jobDescription} Having my record cleared would help me continue to advance in my career.`;
+  // Determine the correct article "a" or "an" based on the first letter of the jobTitle
+  const article = /^[aeiou]/i.test(jobTitle) ? 'an' : 'a';
```
Comment (sydneywalcoff) [inline products/statement-generator/src/helpers/statementGenerators.ts:44]: lets make a 'generateArticle' helper function too.

**Review by sydneywalcoff — CHANGES_REQUESTED** [review]
@davidwiese great work! I'd like to turn the 'article' logic into a 'generateArticle' function and add to the helpers. Should be a pretty quick change.

File: products/statement-generator/src/helpers/statementGenerators.ts (hunk 1 above)
Comment (davidwiese) [reply products/statement-generator/src/helpers/statementGenerators.ts]: All set @sydneywalcoff

**Review by sydneywalcoff — CHANGES_REQUESTED** [review]
@davidwiese thank you for making those changes re: refactoring!! 

I'm unsure if this bug was present last time I checked and I just missed it, but when running locally, I'm not seeing the Capitalization and Punctuation being implemented on some(but not all) of the textarea components. I noticed it primarily on the ones with (2-3 sentences) in the prompt.

**Comment by davidwiese** [comment]
> @davidwiese thank you for making those changes re: refactoring!!
> 
> I'm unsure if this bug was present last time I checked and I just missed it, but when running locally, I'm not seeing the Capitalization and Punctuation being implemented on some(but not all) of the textarea components. I noticed it primarily on the ones with (2-3 sentences) in the prompt.

@sydneywalcoff Hmmm I just ran through it and tested every field possible. I didn't encounter this bug anywhere.

**Comment by davidwiese** [comment]
@sydneywalcoff Added an additional helper to capitalize standalone "I" in the LG form fields

**Review by sydneywalcoff — APPROVED** [review]
@davidwiese this looks great! I re-tested it with manually typed answers and it worked perfectly. Eventually we may want to expand this to also apply when the user edits the responses, but for the time being, I think this is wonderful. Thank you for taking this on!

**Pull request merged**